### PR TITLE
Sam/axl 2642 write up validation api axle docs

### PR DESCRIPTION
**Pull request by Sshatzkin**

### CODE REVIEW

File: docs/api-reference/policies/validation-engine.mdx
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:1]: Would move this to guides/policy-validation.mdx

File: docs/api-reference/policies/validation-engine.mdx [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
+title: "Policy Validation Guide"
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:2]: Just `Policy Validation` (don't need guide)

File: docs/api-reference/policies/validation-engine.mdx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+description: "The Validation Engine is used to perform a list of validations on a policy in order to determine if it meets your business’s requirements."
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:3]: Would just make it actionable. "Perform a set of validations on a policy to determine if your application's requirements are met."

File: docs/api-reference/policies/validation-engine.mdx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+| `fail`    | The policy is not currently active                                    |
+| `unknown` | Not enough information was available to determine the policy's status |
+
+#### Output Schema
+
+```typescript Output Schema
+"policy-active": {
+	"status": "pass",
+	"metadata": {
+		"expDate": "2024-12-25T20:00:00.000Z"
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:48]: This should be checking isActive, not expirationDate and effectiveDate. And we should make sure they keys are the same as in our Policy schema

File: docs/api-reference/policies/validation-engine.mdx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+title: "Policy Validation Guide"
+description: "The Validation Engine is used to perform a list of validations on a policy in order to determine if it meets your business’s requirements."
+---
+
+## Validations
+
+A validation is an individual check run against a
+policy to ensure that it meets a certain requirement. For example, the
+`policy-active` validation checks if the policy is currently active and returns `pass` if it is, and `fail` if it is not.
+or `concern` if it seems to be expired.
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:11]: I would use `unknown` as an example here when isActive: null

File: docs/api-reference/policies/validation-engine.mdx [hunk 5]
```diff
@@ -0,0 +1,187 @@
+---
+title: "Policy Validation Guide"
+description: "The Validation Engine is used to perform a list of validations on a policy in order to determine if it meets your business’s requirements."
+---
+
+## Validations
+
+A validation is an individual check run against a
+policy to ensure that it meets a certain requirement. For example, the
+`policy-active` validation checks if the policy is currently active and returns `pass` if it is, and `fail` if it is not.
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:10]: Would sa y "pass if isActive: true" and "fail if isActive: false". Our audience here are developers, so we can refer to specific keys and technical terminology

File: docs/api-reference/policies/validation-engine.mdx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+or `concern` if it seems to be expired.
+
+Some validations are more complex, providing additional insight which is not contained on the policy, like `rental-covered-for-collision` , which can tell you whether this policy is likely to provide coverage to a driver for collision damage to a rental vehicle.
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:13]: Extra comma + "which provides guidance on whether this policy extends coverage for collision damage when an insured is driving a rental vehicle.

File: docs/api-reference/policies/validation-engine.mdx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+Every validation reports one of the following statuses after running the check:
+
+| Status    | Description                                                                                                                                                                                      |
+| --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `pass`    | The validation has passed entirely                                                                                                                                                               |
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:19]: The policy meets all requirements of the validation

File: docs/api-reference/policies/validation-engine.mdx [hunk 8]
```diff
@@ -0,0 +1,187 @@
+---
+title: "Policy Validation Guide"
+description: "The Validation Engine is used to perform a list of validations on a policy in order to determine if it meets your business’s requirements."
+---
+
+## Validations
+
+A validation is an individual check run against a
+policy to ensure that it meets a certain requirement. For example, the
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:9]: "or set of requirements"

File: docs/api-reference/policies/validation-engine.mdx [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+Some validations are more complex, providing additional insight which is not contained on the policy, like `rental-covered-for-collision` , which can tell you whether this policy is likely to provide coverage to a driver for collision damage to a rental vehicle.
+
+Every validation reports one of the following statuses after running the check:
+
+| Status    | Description                                                                                                                                                                                      |
+| --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `pass`    | The validation has passed entirely                                                                                                                                                               |
+| `fail`    | The policy has been determined to be invalid on this validation                                                                                                                                  |
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:20]: The policy does not meet all requirements of the validation

File: docs/api-reference/policies/validation-engine.mdx [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+## Validations
+
+A validation is an individual check run against a
+policy to ensure that it meets a certain requirement. For example, the
+`policy-active` validation checks if the policy is currently active and returns `pass` if it is, and `fail` if it is not.
+or `concern` if it seems to be expired.
+
+Some validations are more complex, providing additional insight which is not contained on the policy, like `rental-covered-for-collision` , which can tell you whether this policy is likely to provide coverage to a driver for collision damage to a rental vehicle.
+
+Every validation reports one of the following statuses after running the check:
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:15]: Each validation resolves to one of the following statuses:

File: docs/api-reference/policies/validation-engine.mdx [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+| Status    | Description                                                                                                                                                                                      |
+| --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `pass`    | The validation has passed entirely                                                                                                                                                               |
+| `fail`    | The policy has been determined to be invalid on this validation                                                                                                                                  |
+| `caution` | The validation is not entirely `pass` or `fail` and falls somewhere between the two. It's likely that you'll want to engage in a manual review, or handle this according to your risk tolerance. |
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:21]: The policy only partially meets the requirements of the validation or the validation returns an inconclusive outcome.

File: docs/api-reference/policies/validation-engine.mdx [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+| `unknown` | Not enough data was available to determine the outcome of this validation                                                                                                                        |
+
+When making the top level validation determination, the presence of a caution
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:24]: Remove

File: docs/mint.json [hunk 13]
```diff
@@ -92,7 +92,9 @@
           "pages": [
             "api-reference/policies/policy",
             "api-reference/policies/get-policy",
-            "api-reference/policies/get-policy-report"
+            "api-reference/policies/get-policy-report",
+            "api-reference/policies/validation-engine",
+            "api-reference/policies/validate-policy"
```
Comment (niharparikh) [inline docs/mint.json:97]: Move to guides

File: docs/openapi.yml [hunk 14]
```diff
@@ -246,6 +246,157 @@ paths:
...
+                          policy-active:
+                            $ref: "#/components/schemas/ValidationStatus"
+                            description: "The result of the policy-effective validation check."
+                          rental-collision-damage-coverage:
+                            $ref: "#/components/schemas/ValidationStatus"
+                            description: "The result of the rental-collision-damage-coverage validation check."
+                      checks:
+                        type: object
+                        description: "A detailed report of the validation checks."
+                        properties:
```
Comment (niharparikh) [inline docs/openapi.yml:315]: I don't believe you need to specify the exact schema of metadata for each check. I would just keep the required keys: status: enums, metadata: type object with no properties, and breakdown: array of Validation objects

File: docs/api-reference/policies/validation-engine.mdx [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+| coll-does-not-extend-to-rental | `fail` | The policy has collision coverage, but the matching policy agreements indicate that collision coverage does not extend to rental vehicles |
+| coll-not-present | `fail` | The policy does not have collision coverage, and therefore cannot extend collision coverage to rental vehicles |
+
+#### Output Schema
+
+```typescript Output Schema
+"rental-covered-for-collision": {
+  "status": "pass",
+  "message-code": "...",
+  "message": "...",
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:97]: Should be in metadata right?

File: docs/api-reference/policies/validation-engine.mdx [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+| coll-extends-to-rental-with-unknown | `unknown` | The policy has collision coverage, but we do not have enough information to determine if the policy extends collision coverage to rental vehicles |
+| coll-does-not-extend-to-rental | `fail` | The policy has collision coverage, but the matching policy agreements indicate that collision coverage does not extend to rental vehicles |
+| coll-not-present | `fail` | The policy does not have collision coverage, and therefore cannot extend collision coverage to rental vehicles |
+
+#### Output Schema
+
+```typescript Output Schema
+"rental-covered-for-collision": {
+  "status": "pass",
+  "message-code": "...",
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:96]: Should be camelCase: messageCode

File: docs/openapi.yml [hunk 17]
```diff
@@ -246,6 +246,157 @@ paths:
         - Policies
     parameters:
       - $ref: "#/components/parameters/policy"
+  /policies/{id}/validate:
+    "post":
+      summary: Validate Policy
+      description: "The Validate Policy method returns a the result of a series of validation checks on the requested policy object."
```
Comment (niharparikh) [inline docs/openapi.yml:252]: Add note to refer to guide for each validation type

File: docs/api-reference/policies/validation-engine.mdx [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+  "message": "...",
+  "breakdown": {
+    "collision-exists": {
+      "value": "pass",
+      "metadata": {
+        "collision-coverage": {
+          "property": "prp_125940125",
+          "limitPerAccident": 10000,
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:104]: No limitPerAccident, would include the whole coverage object here

File: docs/api-reference/policies/validation-engine.mdx [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+```typescript Output Schema
+"rental-covered-for-collision": {
+  "status": "pass",
+  "message-code": "...",
+  "message": "...",
+  "breakdown": {
+    "collision-exists": {
+      "value": "pass",
+      "metadata": {
+        "collision-coverage": {
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:102]: key here should just be COLL (code of coverage)

File: docs/api-reference/policies/validation-engine.mdx [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+| Message Code | Validation Status | Message |
+| ------------ | ------- | ----- |
+| coll-extends-to-rental | `pass` | The policy has collision coverage and the matching policy agreements indicate that collision coverage extends to rental vehicles |
+| pd-covers-rental-collision | `pass` | The policy is registered in a state in which policies are required to cover collision damage under their property damage terms |
+| coll-extends-to-rental-with-conflict | `caution` | The policy has collision coverage, but the matching policy agreements have conflicting answers about whether collision coverage extends to rental vehicles |
+| coll-extends-to-rental-with-unknown | `unknown` | The policy has collision coverage, but we do not have enough information to determine if the policy extends collision coverage to rental vehicles |
+| coll-does-not-extend-to-rental | `fail` | The policy has collision coverage, but the matching policy agreements indicate that collision coverage does not extend to rental vehicles |
+| coll-not-present | `fail` | The policy does not have collision coverage, and therefore cannot extend collision coverage to rental vehicles |
```
Comment (niharparikh) [inline docs/api-reference/policies/validation-engine.mdx:89]: Will work on this

File: docs/api-reference/policies/validation-engine.mdx (hunk 4 above)
Comment (Sshatzkin) [reply docs/api-reference/policies/validation-engine.mdx:11]: Technically I don't think it's allowed to be null according to our internal schema, but I can add that here just for demonstrative purposes. Internally, this validation is set to just return pass or fail.

File: docs/api-reference/policies/validation-engine.mdx (hunk 19 above)
Comment (Sshatzkin) [reply docs/api-reference/policies/validation-engine.mdx:102]: What do you think of the idea of returning all of the collision coverages here? It's very likely we'll find multiple on a policy and can just return them as an array.

**Review by niharparikh — APPROVED** [review]

**Pull request merged**